app/cdap/components/ConnectionList/index.tsx

```diff
@@ -216,13 +227,37 @@ const DatasetWrapper = () => {
 
   let headerContent;
 
+  const searchHandler = (index: number) => {
+    setDataForTabs((prev) => {
+      const tempData = [...prev];
+      tempData[index].isSearching = true;
+      return tempData;
+    });
+    refs.current[index].focus();
+  };
+
+  const handleChange = (e: any, emitter: string, index: number) => {
+    const val = emitter === 'clear' ? '' : e.target.value.toLowerCase();
```

Comment:
if the two values of emitter are either search or clear then I think i'd prefer two functions handleSearch and handleClear that both wrap another function

Reply:
+1

Reply:
Done. Created two individual functions to handle this search & clear search functionality. 
Now the code is looks more readable.